Large queries get: evaluation exceeded backwards branches

e.g.

```
/home/daurnimator/src/zig/lib/std/mem.zig:159:5: error: evaluation exceeded 1000 backwards branches
    for (source) |s, i|
    ^
./vendor/zig-sqlite/query.zig:100:17: note: called from here
        mem.copy(u8, &parsed_query.query, &buf);
                ^
./vendor/zig-sqlite/sqlite.zig:466:93: note: called from here
    pub fn prepare(self: *Self, comptime query: []const u8) !Statement(.{}, ParsedQuery.from(query)) {
                                                                                            ^
./vendor/zig-sqlite/sqlite.zig:466:71: note: called from here
    pub fn prepare(self: *Self, comptime query: []const u8) !Statement(.{}, ParsedQuery.from(query)) {
                                                                      ^
```

repro:
```
const std = @import("std");

const sqlite = @import("sqlite");

pub fn main() anyerror!void {
    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
    defer arena.deinit();
    const allocator = &arena.allocator;

    var db: sqlite.Db = undefined;
    try db.init(.{
        .mode = sqlite.Db.Mode{ .Memory = {} },
    });

    comptime var i: comptime_int = 0;
    inline while (i < 6) : (i += 1) {
        const query = "select id from users where id != ?{usize}";
        var stmt = try db.prepare(query);
        defer stmt.deinit();

        var iter = try stmt.iterator(usize, .{@as(usize, 3)});

        while (try iter.nextAlloc(allocator, .{})) |r| {
            _ = r;
            // std.log.info("{d} | {s}", .{ r.id, r.name });
        }
    }
}
```
any value above 5 iteration fails with:
```
/home/vincent/dev/perso/libs/zig-sqlite/query.zig:42:13: error: evaluation exceeded 1000 backwards branches
            switch (state) {
            ^
/home/vincent/dev/perso/libs/zig-sqlite/sqlite.zig:466:93: note: called from here
    pub fn prepare(self: *Self, comptime query: []const u8) !Statement(.{}, ParsedQuery.from(query)) {
                                                                                            ^
/home/vincent/dev/perso/libs/zig-sqlite/sqlite.zig:466:71: note: called from here
    pub fn prepare(self: *Self, comptime query: []const u8) !Statement(.{}, ParsedQuery.from(query)) {
                                                                      ^
./src/main.zig:5:29: note: called from here
pub fn main() anyerror!void {
                            ^
/home/vincent/dev/perso/libs/zig-sqlite/sqlite.zig:466:93: note: referenced here
    pub fn prepare(self: *Self, comptime query: []const u8) !Statement(.{}, ParsedQuery.from(query)) {
                                                                                            ^
./src/main.zig:18:34: note: referenced here
        var stmt = try db.prepare(query);
```

not sure if this change in zig is intended or if it's a regression.

--- a/sqlite.zig
+++ b/sqlite.zig
@@ -443,8 +443,11 @@ pub const Db = struct {
     }
 
     /// prepareWithDiags is like `prepare` but takes an additional options argument.
-    pub fn prepareWithDiags(self: *Self, comptime query: []const u8, options: QueryOptions) !Statement(.{}, ParsedQuery.from(query)) {
-        @setEvalBranchQuota(10000);
+    pub fn prepareWithDiags(self: *Self, comptime query: []const u8, options: QueryOptions) !comptime blk: {
+        @setEvalBranchQuota(100000);
+        break :blk Statement(.{}, ParsedQuery.from(query));
+    } {
+        @setEvalBranchQuota(100000);
         const parsed_query = ParsedQuery.from(query);
         return Statement(.{}, comptime parsed_query).prepare(self, options, 0);
     }
@@ -463,8 +466,11 @@ pub const Db = struct {
     /// This is done because we type check the bind parameters when executing the statement later.
     ///
     /// If you want additional error information in case of failures, use `prepareWithDiags`.
-    pub fn prepare(self: *Self, comptime query: []const u8) !Statement(.{}, ParsedQuery.from(query)) {
-        @setEvalBranchQuota(10000);
+    pub fn prepare(self: *Self, comptime query: []const u8) !comptime blk: {
+        @setEvalBranchQuota(100000);
+        break :blk Statement(.{}, ParsedQuery.from(query));
+    } {
+        @setEvalBranchQuota(100000);
         const parsed_query = ParsedQuery.from(query);
         return Statement(.{}, comptime parsed_query).prepare(self, .{}, 0);
     }
@@ -1531,7 +1537,6 @@ test "sqlite: read a single integer value" {
     inline for (types) |typ| {
         const query = "SELECT age FROM user WHERE id = ?{usize}";
 
-        @setEvalBranchQuota(5000);
         var stmt: Statement(.{}, ParsedQuery.from(query)) = try db.prepare(query);
         defer stmt.deinit();
 

